service-base/src/main/java/org/eclipse/hono/service/AbstractApplication.java

```diff
@@ -217,6 +224,9 @@ public final void shutdown(final long maxWaitTime, final Handler<Boolean> shutdo
 
         try {
             preShutdown();
+            if (healthCheckServer != null){
+                healthCheckServer.stopHealthCheckServer();
```

Comment:
IMHO we should wire this up as a precondition to the `vertx.close()` command issued later ...